Add nanodesu-nano, which is more minimalistic and fills gaps to prevent missed clicks

BujuArena opened the pull request:
I've been using the nanodesu theme in XFCE for a couple years now, and while I love the visual style, moving my mouse to the corner of the screen often misses the button I mean to click because there's an unnecessary gap.  This pull request adds `nanodesu-nano`, a fourth theme which changes the title bar height from 28 to 21, filling the click gap, while leaving the position of the icon graphics identical.  This saves screen area and allows the user to quickly click in the corner of a maximized window without missing the click.

Maybe this should be a separate theme instead of replacing the plain nanodesu.  I'm not sure.  This is better to me, at least.

Thanks for creating this theme.

BujuArena committed:
fill gaps in nanodesu to prevent missed clicks

BujuArena committed:
correct border color, 28 height back to 21 height, and thinner title bar to fill gaps instead of fatter buttons

AlexFolland commented:
I added another commit which fills the gaps by thinning the title bar instead of fattening the buttons.  This is more minimal and saves screen area.  It looks and functions the best IMO

BujuArena committed:
restore accidentally-removed helpful comments in themerc

BujuArena committed:
restore original nanodesu (but with correct border colors) and move new changes to nanodesu-nano

AlexFolland commented:
Now I've moved all my changes (except fixing the color of the borders) to a new theme folder, `nanodesu-nano`, so that I'm not overwriting the original theme that someone may want to use.

BujuArena commented:
I made some fundamental changes to this pull request since I originally opened it.  Therefore, I've updated the original message and the title with the final intentions of this pull request and deleted interim comments, for clarity.

BujuArena committed:
set title bar height to 23 and top-left/top-right height to 80 to ease resizing with the mouse

BujuArena committed:
raise close button's X up to its original height

BujuArena committed:
add a pixel column for resizing on the left more easily

BujuArena committed:
final fixes for pixel-perfection with resize handles all around

BujuArena committed:
remove 1-pixel input gap below buttons without ruining look

BujuArena committed:
fix issue when buttons are not present and adopt GTK colors if available

BujuArena committed:
add highlight and press colors (may not be antialiased well)

BujuArena committed:
remove button icon antialiasing to allow using theme colors; also fix corner drag regions so they're all consistent with each other (62 pixels along each corner edge)

BujuArena committed:
reduce corner click regions to 22 by 23 to prevent pushing the window title to the right

BujuArena commented:
I've now also made `nanodesu-nano` adapt to your overall theme's color scheme.  It seemingly works well for all dark themes, but somehow many light themes don't work well.  I'm using Matcha-dark-azul with nanodesu-nano and it looks beautiful with that.

tdloi commented:
@AlexFolland Sorry for late reply and thank you for putting effort into this
However, it's been a long time since I made this, TBH I'm not sure how to process with this anymore so I suggest to upload new theme from your fork instead

BujuArena commented:
Thank you for responding!  I already did upload this to [Pling](https://www.pling.com/p/1670934/).

 I wonder if you still use one of your nanodesu variants.  Do you?  I'm curious.

tdloi commented:
> Thank you for responding! I already did upload this to [Pling](https://www.pling.com/p/1670934/).
> 
> I wonder if you still use one of your nanodesu variants. Do you? I'm curious.

Yes of course, it's one among the first thing I do when reinstalling OS :smile:
![image](https://user-images.githubusercontent.com/33661324/157153710-ab9ebcce-95fa-4988-9ba9-40532fbf8034.png)

BujuArena commented:
Ah, cool!  Yeah, I love the minimalistic visual style that's functional with nice buttons and no wasted screen area.  I like the refinement I added as well.  Have you tried it yet?  If so, what do you think of it?
